pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -458,6 +461,18 @@ public static void validateIngestionAggregation(String name) {
         String.format("aggregation function %s must be one of %s", name, SUPPORTED_INGESTION_AGGREGATIONS));
   }
 
+  static void validateDecoder(StreamConfig streamConfig) {
+    if (streamConfig.getDecoderClass().equals("org.apache.pinot.plugin.inputformat.protobuf.ProtoBufMessageDecoder")) {
+      // check the existence of the needed decoder props
+      if (!streamConfig.getDecoderProperties().containsKey("stream.kafka.decoder.prop.descriptorFile")) {
+        throw new IllegalStateException("Missing property of descriptorFile for ProtoBufMessageDecoder");
+      }
+      if (!streamConfig.getDecoderProperties().containsKey("stream.kafka.decoder.prop.protoClassName")) {
```

Comment:
It this mandatory? Per the implementation, seems it is optional. cc @KKcorps

Reply:
Added this check sepcifically because we ran into very misleading error messages which took us some time to realize this missing field

Reply:
Seems Kartik is OOO.
What is the exception you got without setting this field? We want to make the validation align with the implementation. If you think this setting should be mandatory, we should also change the implementation accordingly

Reply:
The error message is that the protobuf message is corrupted.

Reply:
I think it's better to be stricter, but want to have @KKcorps's thought